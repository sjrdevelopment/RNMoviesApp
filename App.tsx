/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

const Stack = createNativeStackNavigator();

import MovieList from './components/movieList';
import MovieDetail from './components/movieDetail';

function App(): React.JSX.Element {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="MovieList"
          component={MovieList}
          options={{title: 'test'}}
        />
        <Stack.Screen
          name="MovieDetail"
          component={MovieDetail}
          options={{title: 'test'}}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default App;
