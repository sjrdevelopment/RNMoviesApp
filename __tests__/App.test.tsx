/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

import MovieDetail from '../components/movieDetail';

import renderer from 'react-test-renderer';

import ShallowRenderer from 'react-test-renderer/shallow';

// to mock
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {NavigationContainer} from '@react-navigation/native';

// Note: import explicitly to use the types shipped with jest.
import {it, describe, expect, jest} from '@jest/globals';

jest.mock('@react-navigation/native-stack', () => {
  //Mock the default export and named export 'foo'
  return {
    createNativeStackNavigator: () => {
      return {
        navigator: () => {
          return <p>test</p>;
        },
        screen: () => {
          return <p>test</p>;
        },
      };
    },
  };
});

jest.mock('@react-navigation/native', () => {
  //Mock the default export and named export 'foo'
  return {
    NavigationContainer: (children: any) => {
      return <p>test</p>;
    },
  };
});

describe('Matcher style Jest tests', () => {
  // Matcher
  const getColor = (color: string) => {
    return color;
  };

  it('given getColor("red") is called, returns red', () => {
    expect(getColor('red')).toBe('red');
  });
});

describe('Snapshot style Jest tests', () => {
  // Snapshot
  it('renders MovieDetail component correctly', () => {
    const dummyMovie = {
      item: {
        title: 'My Movie',
        releaseYear: '1998',
      },
    };

    const tree = renderer
      .create(
        <MovieDetail
          navigation="test"
          route={{params: dummyMovie}}></MovieDetail>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders App correctly', () => {
    const tree = renderer.create(<App></App>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
