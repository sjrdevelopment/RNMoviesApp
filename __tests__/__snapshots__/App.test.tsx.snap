// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot style Jest tests renders App correctly 1`] = `
<p>
  test
</p>
`;

exports[`Snapshot style Jest tests renders MovieDetail component correctly 1`] = `
<RCTSafeAreaView>
  <Text>
    Movie Detail
  </Text>
  <Text>
    My Movie
  </Text>
  <Text>
    1998
  </Text>
</RCTSafeAreaView>
`;
